packaging_automation/common_tool_methods.py

```diff
@@ -0,0 +1,286 @@
+                    copy_lines[appended_line_index + 1] = append_str
+                    lines_to_be_shifted = lines[line_number + 1:]
+                    copy_lines = copy_lines[0:appended_line_index + 2] + lines_to_be_shifted
+                else:
+                    copy_lines.append(append_str)
+            appended_line_index = appended_line_index + 1
+        edited_content = str_array_to_str(copy_lines)
+    with open(file, "w") as writer:
+        writer.write(edited_content)
+
+    return has_match
+
+
+def prepend_line_in_file(file: str, match_regex: str, append_str: str) -> bool:
+    with open(file, "r+") as reader:
+        file_content = reader.read()
+        lines = file_content.splitlines()
+        has_match = False
+        copy_lines = lines.copy()
+        prepended_line_index = 0
+        for line_number, line in enumerate(lines):
+            if re.match(match_regex, line.strip()):
+                has_match = True
+                copy_lines[prepended_line_index] = append_str
+                lines_to_be_shifted = lines[line_number:]
+                copy_lines = copy_lines[0:prepended_line_index + 1] + lines_to_be_shifted
+            prepended_line_index = prepended_line_index + 1
+        edited_content = str_array_to_str(copy_lines)
+    with open(file, "w") as writer:
+        writer.write(edited_content)
+
+    return has_match
+
+
+def get_current_branch(working_dir: str) -> str:
+    repo = get_new_repo(working_dir)
+    return repo.active_branch
+
+
+def does_remote_branch_exist(branch_name: str, working_dir: str) -> bool:
+    repo = get_new_repo(working_dir)
+    for rp in repo.references:
+        if rp.name.endswith(f"/{branch_name}"):
+            return True
+    return False
+
+
+def does_local_branch_exist(branch_name: str, working_dir: str) -> bool:
+    repo = get_new_repo(working_dir)
+    for rp in repo.branches:
+        if rp.name == branch_name:
+            return True
+    return False
+
+
+def does_branch_exist(branch_name: str, working_dir: str) -> bool:
+    return does_local_branch_exist(branch_name, working_dir) or does_remote_branch_exist(branch_name, working_dir)
+
+
+def get_template_environment(template_dir: str) -> Environment:
+    file_loader = FileSystemLoader(template_dir)
+    env = Environment(loader=file_loader)
+    return env
+
+
+def remove_cloned_code(exec_path: str):
+    release_all_repos()
+    if os.path.exists(f"{exec_path}"):
+        print(f"Deleting cloned code {exec_path} ...")
+        # https://stackoverflow.com/questions/51819472/git-cant-delete-local-branch-operation-not-permitted
+        # https://askubuntu.com/questions/1049142/cannot-delete-git-directory
+        # since git directory is readonly first we need to give write permission to delete git directory
+        if os.path.exists(f"{exec_path}/.git"):
+            run(f"chmod -R 777 {exec_path}/.git")
+        try:
+            run(f"sudo rm -rf {exec_path}")
```

Comment:
Why do we use `sudo` here ? Can we please remove that ?

Reply:
Fixed